sklearn/preprocessing/data.py

```diff
@@ -191,57 +210,33 @@ def fit(self, X, y=None):
             The data used to compute the per-feature minimum and maximum
             used for later scaling along the features axis.
         """
-        X = check_array(X, copy=self.copy, ensure_2d=False)
-        warn_if_not_float(X, estimator=self)
+
+        if sparse.issparse(X):
+            raise TypeError("MinMaxScaler cannot be fitted on sparse inputs")
+
+        if copy is None:
+            copy = self.copy
+
+        X = self._check_array(X, copy)
+
         feature_range = self.feature_range
         if feature_range[0] >= feature_range[1]:
             raise ValueError("Minimum of desired feature range must be smaller"
                              " than maximum. Got %s." % str(feature_range))
         data_min = np.min(X, axis=0)
         data_range = np.max(X, axis=0) - data_min
-        # Do not scale constant features
-        if isinstance(data_range, np.ndarray):
-            data_range[data_range == 0.0] = 1.0
-        elif data_range == 0.:
-            data_range = 1.
+        data_range = self._handle_zeros_in_scale(data_range)
+        self.scalefactor_ = data_range / (feature_range[1] - feature_range[0])
+        self.center_ = data_min - feature_range[0] * self.scalefactor_
+
         self.scale_ = (feature_range[1] - feature_range[0]) / data_range
```

Comment:
If these are here but not used, I guess they should be read-only attributes maybe to make explicit they they are not used?